File: ulyp-ui/src/main/java/com/ulyp/ui/util/TextBuilder.java
package com.ulyp.ui.util;


import javafx.scene.text.Text;

import java.util.ArrayList;
import java.util.List;

public class TextBuilder {

    private String text;
    private List<String> clazzes = new ArrayList<>();

    public TextBuilder text(String text) {
        this.text = text;
        return this;
    }

    public TextBuilder style(String clazz) {
        this.clazzes.add(clazz);
        return this;
    }

    public TextBuilder style(CssClass clazz) {
        this.clazzes.addAll(clazz.getCssClasses());
        return this;
    }

    public Text build() {
        Text te = new Text(text);
        clazzes.forEach(cl -> te.getStyleClass().add(cl));
        return te;
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/util/TextBuilder.kt
package com.ulyp.ui.util

import javafx.scene.text.Text
import java.util.function.Consumer

class TextBuilder {

    private var text: String? = null
    private val clazzes: MutableList<String> = ArrayList()

    fun text(text: String?): TextBuilder {
        this.text = text
        return this
    }

    fun style(clazz: String): TextBuilder {
        clazzes.add(clazz)
        return this
    }

    fun style(clazz: Style): TextBuilder {
        clazzes.addAll(clazz.cssClasses)
        return this
    }

    fun build(): Text {
        val te = Text(text)
        clazzes.forEach(Consumer { cl: String? -> te.styleClass.add(cl) })
        return te
    }
}